TestingLowerBounds/ForMathlib/KernelFstSnd.lean

```diff
@@ -0,0 +1,130 @@
+import Mathlib.Probability.Kernel.MeasureCompProd
+
+open MeasureTheory
+
+open scoped ENNReal
+
+namespace ProbabilityTheory
+
+namespace kernel
+
+variable {α β ι : Type*} {mα : MeasurableSpace α} {mβ : MeasurableSpace β}
+
+variable {γ : Type*} {mγ : MeasurableSpace γ}
+
+open scoped ProbabilityTheory
+
+section FstSnd
+
+variable {δ : Type*} {mδ : MeasurableSpace δ}
+
+
+--TODO: PR this to mathlib, a suitable place may be just under the following lemma
+#check ProbabilityTheory.kernel.snd_swapRight
+
+--I try to reproduce the code that is already there for kernel.fst and kernel.snd, but in this case the fst and snd are referred to the domain of the kernel, not the codomain
+--for now I will use the names fst' and snd', maybe there are better ones
+
+/-- Define a `kernel α γ` from a `kernel (α × β) γ` by taking the comap of TODO:what is this function called?. -/
+noncomputable def fst' (κ : kernel (α × β)  γ) (b : β) : kernel α γ :=
+  comap κ (fun a ↦ (a, b)) (measurable_id.prod_mk measurable_const)
+
+theorem fst'_apply (κ : kernel (α × β)  γ) (b : β) (a : α) : fst' κ b a = κ (a, b) := rfl
+
+@[simp]
+lemma fst'_zero (b : β) : fst' (0 : kernel (α × β)  γ) b = 0 := by simp [fst']
+
+instance IsMarkovKernel.fst' (κ : kernel (α × β)  γ) (b : β) [IsMarkovKernel κ] :
+    IsMarkovKernel (fst' κ b) := by
+  rw [kernel.fst']; infer_instance
+
+instance IsFiniteKernel.fst' (κ : kernel (α × β)  γ) (b : β) [IsFiniteKernel κ] :
+    IsFiniteKernel (fst' κ b) := by
+  rw [kernel.fst']; infer_instance
+
+instance IsSFiniteKernel.fst' (κ : kernel (α × β)  γ) (b : β) [IsSFiniteKernel κ] :
+    IsSFiniteKernel (fst' κ b) := by
+  rw [kernel.fst']; infer_instance
+
+instance (priority := 100) isMarkovKernel_of_isMarkovKernel_fst' {κ : kernel (α × β)  γ}
+    [h : ∀ b, IsMarkovKernel (fst' κ b)] :
+    IsMarkovKernel κ := by
+  refine ⟨fun ab ↦ ⟨?_⟩⟩
+  rw [← fst'_apply]
+  simp only [measure_univ]
+
+--I'm not sure this lemma is actually useful
+lemma comap_fst' (κ : kernel (α × β)  γ) (b : β) {f : δ → α} (hf : Measurable f) :
+    comap (fst' κ b) f hf = comap κ (fun d ↦ (f d, b)) (hf.prod_mk measurable_const) := by
+  ext d s
+  rw [comap_apply, fst'_apply, comap_apply]
+
+@[simp]
+lemma fst'_prodMkLeft (α : Type*) [MeasurableSpace α] (κ : kernel β γ) (a : α) {b : β} :
+    fst' (prodMkLeft α κ) b a = κ b := rfl
+
+@[simp]
+lemma fst'_prodMkRight (β : Type*) [MeasurableSpace β] (κ : kernel α γ) (b : β) :
+    fst' (prodMkRight β κ) b = κ := rfl
+
+/-- Define a `kernel β γ` from a `kernel (α × β) γ` by taking the comap of TODO:what is this function called?. -/
```

Comment:
```suggestion
/-- Define a `kernel β γ` from a `kernel (α × β) γ` by taking the comap of `fun b ↦ (a, b)` for
a given `a : α`. -/
```
That function is also called `Prod.mk` but I think writing it like the above is more explicit.